e3sm_diags/driver/utils/regrid.py

```diff
@@ -0,0 +1,635 @@
+    plevs : List[int] | List[float]
+        A 1-D array of floats or integers representing output pressure levels
+        in mb units. For example, ``plevs=[850.0, 200.0]``. This parameter is
+        usually set by the ``CoreParameter.plevs`` attribute.
+
+    Returns
+    -------
+    xr.Dataset
+        The variable with a Z axis regridded to pressure levels (mb units).
+
+    Notes
+    -----
+    Replaces `e3sm_diags.driver.utils.general.hybrid_to_plevs`.
+    """
+    # TODO: Do we need to convert the Z axis to mb units if it is in PA? Or
+    # do we always expect units to be in mb?
+    ds = dataset.copy()
+
+    z_axis, _ = xc.create_axis("lev", plevs, generate_bounds=False)
+    pressure_grid = xc.create_grid(z=z_axis)
+
+    pressure_coords = _hybrid_to_pressure(ds, var_key)
+
+    # Make sure that the input dataset as Z axis bounds, which are required for
+    # getting grid positions during vertical regridding.
+    try:
+        ds.bounds.get_bounds("Z")
+    except KeyError:
+        ds = ds.bounds.add_bounds("Z")
+
+    result = ds.regridder.vertical(
+        var_key,
+        output_grid=pressure_grid,
+        tool="xgcm",
+        method="log",
+        target_data=pressure_coords,
+    )
+
+    return result
+
+
+def _hybrid_to_pressure(dataset: xr.Dataset, var_key: str) -> xr.DataArray:
+    """Regrid hybrid-sigma levels to pressure coordinates (mb).
+
+    Formula: p(k) = hyam(k) * p0 + hybm(k) * ps
+      * p: pressure data (mb).
+      * hyam: 1-D array equal to hybrid A coefficients.
+      * p0: Scalar numeric value equal to surface reference pressure with
+          the same units as "ps" (mb).
+      * hybm: 1-D array equal to hybrid B coefficients.
+      * ps: 2-D array equal to surface pressure data (mb, converted from Pa).
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        The dataset containing the variable and hybrid levels.
+    var_key : str
+        The variable key.
+
+    Returns
+    -------
+    xr.DataArray
+        The variable with a Z axis on pressure coordinates.
+
+    Raises
+    ------
+    KeyError
+        If the dataset does not contain pressure data (ps) or any of the
+        hybrid levels (hyam, hymb).
+
+    Notes
+    -----
+    This function is equivalent to `geocat.comp.interp_hybrid_to_pressure()`
+    and `cdutil.vertical.reconstructPressureFromHybrid()`.
+    """
+    ds = dataset.copy()
+
+    # p0 is statically set to mb (1000) instead of retrieved from the dataset
+    # because the pressure data should be in mb.
+    p0 = 1000
+    ps = _get_hybrid_sigma_level(ds, "ps")
+    hyam = _get_hybrid_sigma_level(ds, "hyam")
+    hybm = _get_hybrid_sigma_level(ds, "hybm")
+
+    if ps is None or hyam is None or hybm is None:
+        raise KeyError(
+            f"The dataset for '{var_key}' does not contain hybrid-sigma level 'ps', "
+            "'hyam' and/or 'hybm' to use for reconstructing to pressure data."
+        )
+
+    ps = _convert_units_to_mb(ps)
```

Comment:
at least for now all pressure related quantities are brought in/assumed to be in units `mb`

Reply:
I think I pulled `_convert_units_to_mb` over from e3sm_to_cmip just in case units were in `mb`. 

I'll add a TODO to apply this function to the Z axis coordinates too.